### PR TITLE
auth-backend: use externalized microsoft auth implementation again

### PR DESCRIPTION
**Pull request by pjungermann**
This reverts commit 96c4f54bf6070db12676e9af0bf75d0d479c3d72.

PR #20706 fixed the issues that required
the revert of the implementation.

Relates-to: PR #20706
Relates-to: PR #20732
Relates-to: PR #20734
Relates-to: PR #20120
Relates-to: PR #22184

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by pjungermann** [comment]
@Rugvip FYI as you did the revert back then

**Comment by freben** [comment]
@pjungermann How do you feel about the safety of this one - is it better to ship it after tomorrow's release to give it some time to brew in the -next releases to lower the risk? Or is it safe to let out there ASAP?

**Comment by pjungermann** [comment]
@freben I don't think it unsafe. However, I'm fine with keeping this in a "-next" release first. Auth is an important component and we had some trouble with the first attempt :-)

**Review by freben — APPROVED** [review]
Alright let's go

File: plugins/auth-backend/src/providers/microsoft/provider.ts [hunk 1]
```diff
@@ -14,218 +14,25 @@
...
-    }
-  }
-}
+  microsoftAuthenticator,
+  microsoftSignInResolvers,
+} from '@backstage/plugin-auth-backend-module-microsoft-provider';
 
 /**
- * Auth provider integration for Microsoft auth
+ * Auth provider integration for GitLab auth
```
Comment (Rugvip) [inline plugins/auth-backend/src/providers/microsoft/provider.ts:35]:
```suggestion
 * Auth provider integration for Microsoft auth
```

**Review by Rugvip — APPROVED** [review]
Thank you! :+1: 

:shipit:

File: plugins/auth-backend/src/providers/microsoft/provider.ts (hunk 1 above)
Comment (pjungermann) [reply plugins/auth-backend/src/providers/microsoft/provider.ts:35]: oh, seems like the reverted commit still had that. As it's merged already, I will create a tiny PR to take care of it.

**Pull request merged**